modules/xfd.js

```diff
@@ -0,0 +1,2052 @@
+                        tooltip: 'Required. Should not include the ' + Morebits.string.toUpperCaseFirstChar(templateOrModule) + ': namespace prefix.',
+                        required: true
+                    });
+                    target.parentNode.appendChild(tfdtarget.render());
+                }
+                else {
+                    $(Morebits.quickForm.getElementContainer(tfdtarget)).remove();
+                    tfdtarget = null;
+                }
+            },
+            list: [
+                { type: 'option', label: 'Deletion', value: 'tfd', selected: true },
+                { type: 'option', label: 'Merge', value: 'tfm' }
+            ]
+        });
+        this.fieldset.append({
+            type: 'select',
+            name: 'templatetype',
+            label: 'Deletion tag display style: ',
+            tooltip: 'Which <code>type=</code> parameter to pass to the TfD tag template.',
+            list: templateOrModule === 'module' ? [
+                { type: 'option', value: 'module', label: 'Module', selected: true }
+            ] : [
+                { type: 'option', value: 'standard', label: 'Standard', selected: true },
+                { type: 'option', value: 'sidebar', label: 'Sidebar/infobox', selected: !!$('.infobox').length },
+                { type: 'option', value: 'inline', label: 'Inline template', selected: !!$('.mw-parser-output > p .Inline-Template').length },
+                { type: 'option', value: 'tiny', label: 'Tiny inline' }
+            ]
+        });
+        this.fieldset.append({
+            type: 'checkbox',
+            list: [
+                {
+                    label: 'Wrap deletion tag with <noinclude> (for substituted templates only)',
+                    value: 'noinclude',
+                    name: 'noinclude',
+                    tooltip: 'Will wrap the deletion tag in &lt;noinclude&gt; tags, so that it won\'t get substituted along with the template.',
+                    disabled: templateOrModule === 'module',
+                    checked: !!$('.box-Subst_only').length // Default to checked if page carries {{subst only}}
+                }
+            ]
+        });
+        this.appendReasonArea();
+        return this.fieldset;
+    };
+    Tfd.prototype.preprocessParams = function () {
+        if (this.params.tfdtarget) {
+            this.params.tfdtarget = utils.stripNs(this.params.tfdtarget);
+        }
+        // Modules can't be tagged, TfD instructions are to place on /doc subpage
+        this.params.scribunto = mw.config.get('wgPageContentModel') === 'Scribunto';
+        if (this.params.xfdcat === 'tfm') {
+            this.params.otherTemplateName = (this.params.scribunto ? 'Module:' : 'Template:') + this.params.tfdtarget;
+        }
+    };
+    Tfd.prototype.evaluate = function () {
+        var _this = this;
+        var tm = _super.prototype.evaluate.call(this);
```

Comment:
same question about what `tm` means - not going to comment this every time a more informative variable name can be used, but applies elsewhere too